### PR TITLE
Proposing bindHosts and subjectAlternativeNames to LinkAccess CRD

### PR DESCRIPTION
**Pull request by fgiorgetti**

### CODE REVIEW

File: api/types/crds/skupper_link_access_crd.yaml [hunk 1]
```diff
@@ -28,8 +28,14 @@ spec:
                   type: string
                 tlsCredentials:
                   type: string
-                accessType:
-                  type: string
+                bindHosts:
+                  type: array
```
Comment (grs) [inline api/types/crds/skupper_link_access_crd.yaml:32]: Should this be an array or just a single value? Where we want to bind on multiple interfaces, would that not be via separate LinkAccess instances? One part of the role of this resource is to control the listeners in router config. I believe those can only specify one address to bind on? If so, then I think it is nicer to have a separate LinkACcess CR for each distinct address to bind to. (That also allows greater clarity on what names resolve to that interface).

File: pkg/apis/skupper/v1alpha1/types.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ type SkupperClusterPolicySpec struct {
 type Site struct {
 	v1.TypeMeta   `json:",inline"`
 	v1.ObjectMeta `json:"metadata,omitempty"`
-	Spec          SiteSpec `json:"spec,omitempty"`
+	Spec          SiteSpec   `json:"spec,omitempty"`
```
Comment (grs) [inline pkg/apis/skupper/v1alpha1/types.go:45]: These changes seem unrelated to the rest of the PR?

File: go.mod [hunk 3]
```diff
@@ -58,6 +58,7 @@ require (
 	github.com/cespare/xxhash/v2 v2.2.0 // indirect
 	github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect
 	github.com/docker/spdystream v0.0.0-20160310174837-449fdfce4d96 // indirect
+	github.com/emicklei/go-restful v2.9.5+incompatible // indirect
```
Comment (grs) [inline go.mod:61]: Is this a deliberate change?

Comment (fgiorgetti) [reply go.mod:61]: After changing the CRD, I have executed: `FORCE=true scripts/update-codegen.sh`, and I
believe it could have caused those extra mod changes.

File: pkg/apis/skupper/v1alpha1/types.go (hunk 2 above)
Comment (fgiorgetti) [reply pkg/apis/skupper/v1alpha1/types.go:45]: These are probably a result of `go fmt` that I ran locally.

File: api/types/crds/skupper_link_access_crd.yaml (hunk 1 above)
Comment (fgiorgetti) [reply api/types/crds/skupper_link_access_crd.yaml:32]: I was actually in doubt about it as well. Thinking of eventual usages on kube sites, I believe it makes more
sense to have a single host. I will update it.

**Review by grs — APPROVED** [review]
